src/lib.rs

```diff
@@ -671,6 +671,14 @@ impl Config {
                     panic!("unsupported darwin target: {}", target);
```

Comment:
This would need to be updated to support the other architectures used on Apple platforms, namely `i686`, `arm64e`, `i386`, `armv7s`, `armv7k` and `arm64_32` (all of which should be passed with the same name to CMake).